core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SecurityScanWorkflowDetectProbe.java

```diff
@@ -0,0 +1,97 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+@Component
+@Order(value = SecurityScanWorkflowDetectProbe.ORDER)
+public class SecurityScanWorkflowDetectProbe extends Probe {
+    public static final int ORDER = LastCommitDateProbe.ORDER + 100;
+    public static final String KEY = "securityscanworkflowdetect";
+
+    private static final String SEARCH_LINE = "uses: jenkins-infra/jenkins-security-scan/.github/workflows/jenkins-security-scan.yaml@v2";
+    private static final String WORKFLOWS_DIRECTORY = ".github/workflows";
+
+    @Override
+    protected ProbeResult doApply(Plugin plugin, ProbeContext context) {
+        final Path repository = context.getScmRepository();
+        final Path workflowsPath = Paths.get(repository.toString(), WORKFLOWS_DIRECTORY);
+        try (Stream<Path> paths = Files.walk(workflowsPath)) {
+            List<String> fileNames = new ArrayList<>();
+            paths.filter(Files::isRegularFile)
+                .filter(this::containsSearchLine)
+                .forEach(path -> fileNames.add(path.toString()));
+            if (!fileNames.isEmpty()) {
+                String fileList = String.join(", ", fileNames);
+                return ProbeResult.success(key(), "The line is present in the following files: " + fileList);
+            } else {
+                return ProbeResult.failure(key(), "The line is not present in any file in the repository");
+            }
+        } catch (IOException e) {
+            return ProbeResult.error(key(), e.getMessage());
+        }
+    }
+
+    private boolean containsSearchLine(Path file) {
+        try (Stream<String> lines = Files.lines(file)) {
+            return lines.anyMatch(line -> line.contains(SEARCH_LINE));
+        } catch (IOException e) {
+            return false;
+        }
+    }
```

Comment:
Please see #315 (now merged) and extract the `WorkflowDefinition` and `WorkflowJobDefinition` into the package `io.jenkins.pluginhealth.scoring.model` so that you can use them here as well.

Reply:
Hey Adrien, can you please elaborate over this about what to do? I am not able to understand.

Reply:
in #315, I created 2 `record` to parse the GitHub Actions workflow definitions easily (https://github.com/jenkins-infra/plugin-health-scoring/blob/983d8f8fc817abb5d0d78b76ed52b74226b74d75/core/src/main/java/io/jenkins/pluginhealth/scoring/probes/ContinuousDeliveryProbe.java#L83-L89).
Currently, as those record are inside the class, you cannot use them in your PR. But you can move them into the `io.jenkins.pluginhealth.scoring.model` package and then you can use them.